src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1338 @@
+        self.speech_model.token_embedding = value
+
+    @add_start_docstrings_to_model_forward(CLVP_SPEECH_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=CLVPSpeechConfig)
+    def forward(
+        self,
+        speech_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        use_causal_attention_mask: Optional[bool] = False,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutputWithPooling]:
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> import torch
+        >>> from transformers import CLVPSpeechModel
+
+        >>> model = CLVPSpeechModel.from_pretrained("susnato/clvp_dev")
+
+        >>> # TODO : after FeatureExtractor is implemented we need to change it to something lik FE.__call__(...)
+        >>> inputs = {"speech_ids": torch.tensor([[56, 8, 48, 7, 11, 23]]).long()}
+
+        >>> outputs = model(**inputs)
+        >>> last_hidden_state = outputs.last_hidden_state
+        >>> pooled_output = outputs.pooler_output
+        ```
+        """
+
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        return self.speech_model(
+            input_ids=speech_ids,
+            attention_mask=attention_mask,
+            use_causal_attention_mask=use_causal_attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+
+@add_start_docstrings(CLVP_START_DOCSTRING)
+class CLVPModel(CLVPPreTrainedModel):
+    config_class = CLVPConfig
+
+    def __init__(self, config: CLVPConfig):
+        super().__init__(config)
+
+        if not isinstance(config.text_config, CLVPTextConfig):
+            raise ValueError(
+                "config.text_config is expected to be of type CLVPTextConfig but is of type"
+                f" {type(config.text_config)}."
+            )
+
+        if not isinstance(config.speech_config, CLVPSpeechConfig):
+            raise ValueError(
+                "config.speech_config is expected to be of type CLVPSpeechConfig but is of type"
+                f" {type(config.speech_config)}."
+            )
+
+        text_config = config.text_config
+        speech_config = config.speech_config
+
+        self.projection_dim = config.projection_dim
+        self.text_embed_dim = text_config.hidden_size
+        self.speech_embed_dim = speech_config.hidden_size
+
+        self.text_model = CLVPTransformer(text_config)
+        self.speech_model = CLVPTransformer(speech_config)
+
+        self.text_projection = nn.Linear(self.text_embed_dim, self.projection_dim, bias=False)
+        self.speech_projection = nn.Linear(self.speech_embed_dim, self.projection_dim, bias=False)
```

Comment:
Here, you could instead instantiate from `CLVPTransformerWithProjection` if you decide to goes with it. IMO, it would make the code way a bit easier to read!

```suggestion
        self.text_model = CLVPTransformerWithProjection(text_config, self.projection_dim, bias=False)
        self.speech_model = CLVPTransformerWithProjection(speech_config, self.projection_dim, bias=False)
```